pkg/server/container_create.go

```diff
@@ -251,7 +252,10 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 
 	opts = append(opts,
 		containerd.WithSpec(spec, specOpts...),
-		containerd.WithRuntime(defaultRuntime, nil),
+		containerd.WithRuntime(c.config.ContainerdConfig.Runtime,
+			&runcopts.RuncOptions{Runtime: c.config.ContainerdConfig.RuntimeEngine,
+				RuntimeRoot:   c.config.ContainerdConfig.RuntimeRoot,
```

Comment:
`c.config.ContainerdRuntimeRoot`, and also for the other configurations.

Reply:
are you asking to move these options out of the containerdconfig struct? and prefix them with containerd instead?

Reply:
Nope. `ContainerdConfig` is embedded in Config, you could access its field directly.